smart-contract-sale/README.md

```diff
@@ -0,0 +1,285 @@
+    --keyring-backend test \
+    --home build/node0 \
+    --chain-id chain-local \
+    --gas auto --gas-prices 1905nhash --gas-adjustment 1.4 \
+    --broadcast-mode block \
+    --yes \
+    --testnet -o json | jq
+```
+
+## Instantiate the Sale Smart Contract
+
+Instantiate the contract with the following params:
+
+- DCC Address: `tp14hj2tavq8fpesdwxxcu44rty3hh90vhujrvcmstl4zr3txmfvw9s96lrg8`
+- DCC Denom: `usdf.local`
+
+```bash
+provenanced tx wasm instantiate 2 '{"dcc_address":"tp14hj2tavq8fpesdwxxcu44rty3hh90vhujrvcmstl4zr3txmfvw9s96lrg8","dcc_denom":"usdf.local"}' \
+    --admin $(provenanced keys show -a node0 --keyring-backend test --home build/node0 --testnet) \
+    --label dcc_sale_poc_v1 \
+    --from node0 \
+    --keyring-backend test \
+    --home build/node0 \
+    --chain-id chain-local \
+    --gas auto --gas-prices 1905nhash --gas-adjustment 2 \
+    --broadcast-mode block \
+    --yes \
+    --testnet -o json | jq
+```
+
+## Add Executor to DCC Smart Contract
+
+Admin of DCC smart contract must add the smart contract address to list of allowable executors so that this instance can complete
+sales by making transfer calls on behalf of buyer.
+
+```bash
+provenanced tx wasm execute \
+    tp14hj2tavq8fpesdwxxcu44rty3hh90vhujrvcmstl4zr3txmfvw9s96lrg8 \
+    '{"add_executor":{"id":"tp1nc5tatafv6eyq7llkr2gv50ff9e22mnf70qgjlv737ktmt4eswrqf06p2p"}}' \
+    --from node0 \
+    --keyring-backend test \
+    --home build/node0 \
+    --chain-id chain-local \
+    --gas auto --gas-prices 1905nhash --gas-adjustment 2 \
+    --broadcast-mode block \
+    --yes \
+    --testnet -o json | jq
+```
+
+## Create Sale
+
+Let's assume `user1` wants to sell 1 `hash` for 1,000 `usdf.local` to `user2`.
+
+```bash
+provenanced tx wasm execute \
+    tp1nc5tatafv6eyq7llkr2gv50ff9e22mnf70qgjlv737ktmt4eswrqf06p2p \
+    '{"create_sale":{"id":"b476aca6-bd69-4e54-b123-614b116321d3","price":{"amount":"1000","denom":"usdf.local"},"buyer":"tp1m4arun5y9jcwkatq2ey9wuftanm5ptzsg4ppfs"}}' \
+    --amount '1000000000nhash' \
+    --from user1 \
+    --keyring-backend test \
+    --home build/node0 \
+    --chain-id chain-local \
+    --gas auto --gas-prices 1905nhash --gas-adjustment 2 \
+    --broadcast-mode block \
+    --yes \
+    --testnet -o json | jq
+```
+
+Query the sale proposal to see the order is now available and in a pending status.
+
+```bash
+provenanced query wasm contract-state smart tp1nc5tatafv6eyq7llkr2gv50ff9e22mnf70qgjlv737ktmt4eswrqf06p2p \
+    '{"get_sale":{"id":"b476aca6-bd69-4e54-b123-614b116321d3"}}' \
+    --ascii \
+    -o json \
+    --chain-id chain-local -t | jq
+{
+  "data": {
+    "id": "b476aca6-bd69-4e54-b123-614b116321d3",
+    "asset": {
+      "denom": "nhash",
+      "amount": "1000000000"
+    },
+    "owner": "tp10nnm70y8zc5m8yje5zx5canyqq639j3ph7mj8p",
+    "buyer": "tp1m4arun5y9jcwkatq2ey9wuftanm5ptzsg4ppfs",
+    "price": {
+      "denom": "usdf.local",
+      "amount": "1000"
+    },
+    "status": "pending"
```

Comment:
one thing i'm not sure of the scalability of is if it makes sense to keep completed/canceled sales in state vs removing them

for some initial test transactions, keeping state is probably okay